File: iosApp/iosApp/Services/DegenServiceAssembler.swift
// Created by web3d4v on 16/05/2022.
// Copyright (c) 2022 Sons Of Crypto.
// SPDX-License-Identifier: MIT

import web3walletcore

final class DegenServiceAssembler: AssemblerComponent {
    
    func register(to registry: AssemblerRegistry) {
        
        registry.register(scope: .singleton) { resolver -> DegenService in
            
            DefaultDegenService(
                walletService: resolver.resolve()
            )
        }
    }
}

File: androidApp/src/main/java/com/sonsofcrypto/web3wallet/android/services/DegenServiceAssembler.kt
package com.sonsofcrypto.web3wallet.android.services

import com.sonsofcrypto.web3lib.services.wallet.WalletService
import com.sonsofcrypto.web3wallet.android.common.AssemblerComponent
import com.sonsofcrypto.web3wallet.android.common.AssemblerRegistry
import com.sonsofcrypto.web3wallet.android.common.AssemblerRegistryScope
import com.sonsofcrypto.web3walletcore.services.degen.DefaultDegenService
import com.sonsofcrypto.web3walletcore.services.degen.DegenService
import smartadapter.internal.extension.name

class DegenServiceAssembler: AssemblerComponent {

    override fun register(to: AssemblerRegistry) {

        to.register(DegenService::class.name, AssemblerRegistryScope.SINGLETON) {
            DefaultDegenService(
                it.resolve(WalletService::class.name)
            )
        }
    }
}